TeapotTests/TeapotTests.swift

```diff
@@ -78,6 +78,64 @@ class TeapotTests: XCTestCase {
         self.waitForExpectations(timeout: 20.0)
     }
 
+    func testPutWithJSONData() {
+        let expectation = self.expectation(description: "Put with JSON")
+
+        let dict = "{\"foo\":\"bar\"}"
```

Comment:
I guess it's not necessary to do this test for array data since they are ordered? Just wonder if we should ad this test since the code has changed for array as well as dicts.

Reply:
That's a really good idea - I'll add another for arrays.